src/sezzle-checkout-button/sezzle-checkout-button-asset.js

```diff
@@ -97,3 +97,41 @@ class SezzleCheckoutButton {
 		this.createButton()
 	}
 }
+
+class EventLogger {
+	constructor(options){
+		this.merchantUUID = options.merchantUUID || '';
+	}
+
+	sendEvent(eventName, description="") {
```

Comment:
As this method is going to be used in the `catch` block up there in the initialization script, I'd suggest to also wrap it in `try..catch` and maybe console log it with a bit extra info that we failed to log an event.

Reply:
`Promise` already has a `try...catch` around it and not throwing the error, so adding the catch there. I copied this section from sezzle-js, so we need to address it there as well.

Reply:
I didn't notice the `.catch` with the console logging. In this case, I think we're good without the try...catch in the `sendEvent`.

Not sure about sezzle-js, this is the `logEvent` from sezzle-js `eventLogger`:

` logEvent(eventName, eventDescription) {
    try {
      const event = { name: eventName, description: eventDescription ?? "" };
      // If tracking is disabled, log only errors and warnings
      if (!this.#noTracking || event.name === "error" || event.name === "warning") {
        this.#events.push(event);
      }
      // Push events only on errors (request finished with error), on loads (request finished with success) or on clicks (after load events)
      if (event.name === "error" || event.name === "onload" || event.name.startsWith("onclick")) {
        this.#sendEvents();
        this.#events = [];
      }
    } catch (error) {
      console.log(error);
    }
  }`